common/src/main/java/com/microsoft/identity/common/internal/net/UrlConnectionHttpClient.java

```diff
@@ -22,325 +22,186 @@
-        Telemetry.emit(
-                new HttpStartEvent()
-                        .putMethod(requestMethod)
-                        .putPath(requestUrl)
-                        .putRequestIdHeader(requestId)
-        );
-    }
+        public Builder connectTimeoutMsSupplier(@NonNull final Supplier<Integer> connectTimeoutMsSupplier) {
+            mConnectTimeoutMsSupplier = connectTimeoutMsSupplier;
+            return this;
+        }
 
-    /**
-     * Record the end of an http event.
-     * @param response
-     */
-    private static void recordHttpTelemetryEventEnd(@Nullable final HttpResponse response) {
-        final HttpEndEvent httpEndEvent = new HttpEndEvent();
+        public Builder readTimeoutMsSupplier(@NonNull final Supplier<Integer> readTimeoutMsSupplier) {
+            mReadTimeoutMsSupplier = readTimeoutMsSupplier;
+            return this;
+        }
 
-        if (null != response) {
-            httpEndEvent.putStatusCode(response.getStatusCode());
+        public Builder streamBufferSize(final int streamBufferSize) {
+            mStreamBufferSize = streamBufferSize;
+            return this;
         }
 
-        Telemetry.emit(httpEndEvent);
+        public UrlConnectionHttpClient build() {
+            return new UrlConnectionHttpClient(mRetryPolicy,
+                    mConnectTimeoutMs,
+                    mReadTimeoutMs,
+                    mConnectTimeoutMsSupplier,
+                    mReadTimeoutMsSupplier,
+                    mStreamBufferSize);
+        }
     }
 
-    /**
-     * Sends an HTTP request of the specified method; applies appropriate provided arguments where
-     * applicable.
-     *
-     * @param httpMethod         One of: GET, POST, HEAD, PUT, DELETE, TRACE, OPTIONS, PATCH.
-     * @param requestUrl         The recipient {@link URL}.
-     * @param requestHeaders     Headers used to send the http request.
-     * @param requestContent     Optional request body, if applicable.
-     * @return HttpResponse      The response for this request.
-     * @throws IOException If an error is encountered while servicing this request.
-     */
-    @Override
-    public HttpResponse method(@NonNull final HttpMethod httpMethod,
-                               @NonNull final URL requestUrl,
-                               @NonNull final Map<String, String> requestHeaders,
-                               @Nullable final byte[] requestContent) throws IOException {
-        recordHttpTelemetryEventStart(httpMethod.name(), requestUrl, requestHeaders.get(CLIENT_REQUEST_ID));
-        final HttpRequest request = constructHttpRequest(httpMethod, requestUrl, requestHeaders, requestContent);
-        return retryPolicy.attempt(new Callable<HttpResponse>() {
-            public HttpResponse call() throws IOException {
-                return executeHttpSend(request, new Consumer<HttpResponse>() {
+    private com.microsoft.identity.common.java.net.UrlConnectionHttpClient sAdaptedObject;
```

Comment:
This does not seem static

```suggestion
    private com.microsoft.identity.common.java.net.UrlConnectionHttpClient mClientInstance;
```

Reply:
It's not :)